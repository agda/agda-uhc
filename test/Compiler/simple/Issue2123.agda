module _ where

open import Agda.Builtin.Nat
open import Agda.Builtin.String
open import Common.IO
open import Common.Unit

data ⊥ : Set where

record Pair : Set where
  constructor _,_
  field fst snd : String

open Pair

record ERing : Set where
  constructor ering
  field divRem : (⊥ → ⊥) → Pair

open ERing

eRing : ERing
eRing = ering λ _ → "fst" , "snd"

test : String
test = snd (divRem eRing (λ ()))

main : IO Unit
main = putStrLn test
