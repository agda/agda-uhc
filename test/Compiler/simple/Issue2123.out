EXECUTED_PROGRAM

ret > ExitSuccess
out > snd
out >
